src/libmongoc/src/mongoc/mongoc-topology-description.c

```diff
@@ -678,11 +735,18 @@ mongoc_topology_description_suitable_servers (
    const mongoc_set_t *td_servers = mc_tpld_servers_const (topology);
    int64_t nearest = -1;
    int i;
-   mongoc_read_mode_t read_mode = mongoc_read_prefs_get_mode (read_pref);
+   const mongoc_read_mode_t given_read_mode =
+      mongoc_read_prefs_get_mode (read_pref);
+   const mongoc_read_mode_t effective_read_mode =
+      _calc_effective_read_mode (topology, optype, given_read_mode);
+
+   if (chosen_read_mode) {
+      *chosen_read_mode = effective_read_mode;
+   }
 
    candidates = bson_malloc0 (sizeof (*candidates) * td_servers->items_len);
 
-   data.read_mode = read_mode;
+   data.read_mode = effective_read_mode;
```

Comment:
There is mixed usage of both `data.read_mode` and `effective_read_mode` in this function following this assignment, despite both being guaranteed to have the same value. This creates a misleading impression that it matters which is being used. For clarity, can the function be updated to use only one or the other following this assignment?

Reply:
Yeah, this is confusing. I can clean things up by moving most information into `data`.